cpp/box/Box.h

```diff
@@ -435,6 +446,25 @@ class Box
         );
     }
 
+    //! Get mask of points that fit inside the box.
+    /*! \param points Point positions.
+        \param n_points The number of points.
+        \param cropped_mask Mask of points inside the box.
+    */
+    void crop(const vec3<float>* points, const unsigned int n_points,
+        bool* cropped_mask) const
+    {
+        util::forLoopWrapper(0, n_points, [&](size_t begin, size_t end) {
+            for (size_t i = begin; i < n_points; ++i)
+            {
+                std::transform(&points[begin], &points[end], &cropped_mask[begin], 
+                [this](vec3<float> point) -> bool {
```

Comment:
```suggestion
                [this](const vec3<float> &point) -> bool {
```

Reply:
It looks like `std::transform` requires an output. For a binary operations, it should take the form:
```cpp
transform(Iterator inputBegin1, Iterator inputEnd1, 
         Iterator inputBegin2, Iterator OutputBegin, 
         binary_operation) 
```
I'm not sure what to use as an output iterator when the function is void. I saw something about `void` pointers but I'm not sure that's relevant to this problem. There are some iterators like `std::for_each` but that only take unary operators so I'm not sure they're an appropriate replacement for `std::transform` in this case. Are you aware of any alternate solutions?

Reply:
I didn't change the output of the function, I just suggested passing the input by reference to avoid making a copy of it. Also I'm not sure what you mean about the binary operator, this is a unary operator (in terms of number of iterators, see the first signature on [this page I assume you are referring to](https://en.cppreference.com/w/cpp/algorithm/transform)) since you're just comparing everything to 0.

Reply:
Actually... Just ignore my comment. I was just confused.